src/flavors/arc_swap.rs

```diff
@@ -3,21 +3,26 @@ use crate::{async_publisher, async_subscriber, publisher, subscriber, SwapSlot};
 use arc_swap::ArcSwapOption;
 use std::sync::Arc;
 
-pub type Slot<T> = ArcSwapOption<T>;
+pub struct Slot<T> {
```

Comment:
The idiomatic way to wrap one struct with another in Rust is to use `struct Slot<T>(ArcSwapOption);`.
This equates to Slot being a tuple with only one member.

To implement various methods.
```
fn load(&self) -> Option<Arc<T>> {
    self.0.load_full()
}
```
0 is used to access the first member of a tuple.

Reply:
Why do you think that's the idiomatic way to wrap an object? Look for an example: https://github.com/jdavis/rust-design-patterns/blob/master/patterns/adapter.rs  

I like it more this way, because it's much more readable than with `0` access attribute.

Reply:
too minor of deal to care :D,
use struct if you prefer them.